debugger/src/main/java/org/codice/acdebugger/api/SecuritySolution.java

```diff
@@ -0,0 +1,192 @@
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/** Represents a solution to a security failure that was detected while debugging. */
+public class SecuritySolution implements Comparable<SecuritySolution> {
+  /** The failed permissions info */
+  protected final Set<String> permissionInfos;
+
+  /** Bundles that are granted the permission */
+  protected final Set<String> grantedBundles;
+
+  /** List of locations where to add doPrivileged() blocks */
+  protected final List<StackFrameInformation> doPrivileged;
+
+  /**
+   * Constructs a solution.
+   *
+   * @param permissionInfos the failed permissions info
+   * @param grantedBundles bundles that are granted the permission
+   * @param doPrivileged list of locations where to add doPrivileged() blocks
+   */
+  public SecuritySolution(
+      Set<String> permissionInfos,
+      Set<String> grantedBundles,
+      List<StackFrameInformation> doPrivileged) {
+    this.permissionInfos = permissionInfos;
+    this.grantedBundles = grantedBundles;
+    this.doPrivileged = doPrivileged;
+  }
+
+  /**
+   * Constructs a solution.
+   *
+   * @param solution the original solution to clone
+   */
+  protected SecuritySolution(SecuritySolution solution) {
+    this.permissionInfos = solution.permissionInfos;
+    this.grantedBundles = new HashSet<>(solution.grantedBundles);
+    this.doPrivileged = new ArrayList<>(solution.doPrivileged);
+  }
+
+  /**
+   * Gets the permissions associated with this security solution.
+   *
+   * @return the permissions associated with this security solution
+   */
+  public Set<String> getPermissions() {
+    return Collections.unmodifiableSet(permissionInfos);
+  }
+
+  /**
+   * Gets the bundles that have to be given permissions for this solution.
+   *
+   * @return bundles requiring permissions for this solution
+   */
+  public Set<String> getGrantedBundles() {
+    return Collections.unmodifiableSet(grantedBundles);
+  }
+
+  /**
+   * Gets the locations where to introduce <code>doPrivileged()</code> blocks for this solution.
+   *
+   * @return the locations where to introduce <code>doPrivileged()</code> blocks for this solution
+   */
+  public List<StackFrameInformation> getDoPrivilegedLocations() {
+    return Collections.unmodifiableList(doPrivileged);
+  }
+
+  /** Prints information about this solution. */
+  public void print() {
+    print("");
+  }
+
+  /**
+   * Prints information about this solution.
+   *
+   * @param prefix a prefix string to prepend to each printed line
+   */
+  @SuppressWarnings("squid:S106" /* this is a console application */)
+  public void print(String prefix) {
+    System.out.println(prefix + "{");
+    final String and;
+
+    if (!grantedBundles.isEmpty()) {
+      final String s = (permissionInfos.size() == 1) ? "" : "s";
+
```

Comment:
❓ Will we always be adding to default.policy?

Reply:
Updated as per your previous comment.